plugins/outputs/nats/nats_test.go

```diff
@@ -30,12 +33,19 @@ func TestConnectAndWriteIntegration(t *testing.T) {
 	serializer := &influx.Serializer{}
 	require.NoError(t, serializer.Init())
 	n := &NATS{
-		Servers:    server,
-		Name:       "telegraf",
-		Subject:    "telegraf",
+		Servers: server,
+		Name:    "telegraf",
+		Subject: "telegraf",
+		Jetstream: &JetstreamConfig{
+			AutoCreateStream: true,
+			StreamConfig: jetstream.StreamConfig{
+				Name:     "my-telegraf-stream",
+				Subjects: []string{"telegraf"},
+			},
+		},
 		serializer: serializer,
 	}
-
+	time.Sleep(3 * time.Second)
```

Comment:
Please no sleeps :) Is there something additional we can use require.Eventually to check the server for? or is there a different log message to look for?

Reply:
I have used a different wait strategy. It would wait for port. This was working reliably for me on my setup.